include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -295,51 +282,20 @@ class alignment_algorithm :
         // Cleanup and optionally compute the traceback.
         // ----------------------------------------------------------------------------
 
-        using result_t = typename align_result_selector<first_range_t, second_range_t, config_t>::type;
-        result_t res{};
-
-        res.id = idx;
-        // Balance the score with possible leading/trailing gaps depending on the
-        // band settings.
-        this->balance_leading_gaps(get<3>(cache), band, gap);
+        return alignment_result{make_result(idx, trimmed_first_range, trimmed_second_range, cache)};
+    }
 
-        this->balance_trailing_gaps(get<3>(cache),
-                                    this->dimension_first_range,
-                                    this->dimension_second_range,
-                                    band,
-                                    gap);
+private:
 
-        if constexpr (config_t::template exists<align_cfg::result<detail::with_score_type>>())
-        {
-            res.score = get<3>(cache).score;
-        }
-        if constexpr (config_t::template exists<align_cfg::result<with_back_coordinate_type>>())
-        {
-            res.score = get<3>(cache).score;
-            res.back_coordinate = this->map_banded_coordinate_to_range_position(get<3>(cache).coordinate);
-        }
-        if constexpr (config_t::template exists<align_cfg::result<with_front_coordinate_type>>())
-        { // At the moment we also compute the traceback even if only the front coordinate was requested.
-          // This can be later optimised by computing the reverse alignment in linear memory from the maximum.
-          // Especially for the SIMD version this might be more efficient.
-            res.score = get<3>(cache).score;
-            res.back_coordinate = this->map_banded_coordinate_to_range_position(get<3>(cache).coordinate);
-            res.front_coordinate =
-                get<0>(compute_traceback(first_range,
-                                         second_range,
-                                         get<3>(cache).coordinate));
-        }
-        if constexpr (config_t::template exists<align_cfg::result<with_alignment_type>>())
-        {
-            res.score = get<3>(cache).score;
-            res.back_coordinate = this->map_banded_coordinate_to_range_position(get<3>(cache).coordinate);
-            std::tie(res.front_coordinate, res.alignment) = compute_traceback(first_range,
-                                                                              second_range,
-                                                                              get<3>(cache).coordinate);
-        }
-        return alignment_result{res};
+    //!\brief Initialises the debug matrices for the given sequences.
+    template <typename first_range_t, typename second_range_t>
+    constexpr void initialise_debug_matrices(first_range_t & first_range, second_range_t & second_range)
```

Comment:
`&` because I do not need to forward rvalues inside of the alignment algorithm. Not `const &` because I don't care if the sequence is `const_iterable` or not.

Reply:
I hate this non const_iterablity of views :roll_eyes:.